evap/static/ts/src/quick-review-slider.ts

```diff
@@ -0,0 +1,410 @@
+        this.startOverTriggers.undecided.classList.toggle("d-none", allAreReviewed);
+        this.startOverTriggers.all.classList.toggle("d-none", !slidesExist);
+    };
+
+    //
+    // Sliding
+    //
+    public startOver = (where: StartOverWhere) => {
+        const decided = this.slider.querySelectorAll<HTMLElement>(`[data-layer="${Layer.Answer}"][data-review]`);
+        const undecided = this.slider.querySelectorAll<HTMLElement>(
+            `[data-layer="${Layer.Answer}"]:not([data-review])`,
+        );
+        this.answerSlides = Array.from(decided).concat(Array.from(undecided));
+
+        const startIndex = where === StartOverWhere.Undecided && undecided.length > 0 ? decided.length : 0;
+        this.slideTo(startIndex);
+        this.updateButtons();
+    };
+    public slideTo = (requestedIndex: number) => {
+        requestedIndex = clamp(requestedIndex, 0, this.answerSlides.length);
+        const direction = requestedIndex >= this.selectedSlideIndex ? "right" : "left";
+        this.selectedSlideIndex = requestedIndex;
+
+        let nextActiveElement;
+        if (this.isShowingEndslide()) {
+            nextActiveElement = this.alertSlide;
+            this.updateAlertSlide();
+        } else {
+            nextActiveElement = this.selectedSlide;
+        }
+        this.slideLayer(Layer.Answer, direction, nextActiveElement);
+        this.updateAnswerIdInputs();
+
+        this.updateNavigationButtons();
+    };
+    private slideLayer = (layer: Layer, direction: SlideDirection, nextActiveElement?: HTMLElement) => {
+        // to preserve the vertical positions and heights during transition,
+        // the elements will be deactivated from bottom to top and activated from top to bottom
+
+        if (nextActiveElement?.classList.contains("active")) {
+            return;
+        }
+        const last = this.slider.querySelector<HTMLElement>(`[data-layer="${layer}"].active, .alert.active`);
+        if (last) {
+            const reversed: SlideDirection = direction === "left" ? "right" : "left";
+            last.style.top = `${last.offsetTop}px`;
+            last.style.height = `${last.getBoundingClientRect().height}px`;
+            last.classList.remove("active", "to-left", "to-right");
+            last.classList.add(`to-${reversed}`);
+        }
+
+        if (layer > 0) {
+            let activeInParentLayer;
+            if (nextActiveElement && !this.isShowingEndslide()) {
+                activeInParentLayer =
+                    findPreviousElementSibling(nextActiveElement, `[data-layer="${layer - 1}"]`) ?? undefined;
+                activeInParentLayer = activeInParentLayer as HTMLElement | undefined;
+            }
+            this.slideLayer(layer - 1, direction, activeInParentLayer);
+        }
+
+        if (nextActiveElement) {
+            // First, translate nextActiveElement to the left or right and
+            // then, in the next frame, move it back to the middle
```

Comment:
I think we want to have a bit more explanation here. Especially: The animation takes more than one browser frame. Why is it okay to change the class in the next frame?